docs/playground/src/PlaygroundConfiguration.ts

```diff
@@ -278,61 +376,107 @@ export const PlaygroundConfiguration: IStringMap<IComponentPlaygroundConfigurati
-    show: true,
-    toExecute: ()=> {
-      $$(document.body).find('.CoveoSearchInterface').style.minHeight = '300px';
-    }
-  },
-  SearchAlerts: {
-    show: true,
-    element: $$('div', undefined, `<div class="CoveoSettings"></div><div class="CoveoSearchbox"></div><div class="CoveoSearchAlerts"></div>`),
-    toExecute: ()=> {
-      $$(document.body).find('.CoveoSearchInterface').style.minHeight = '300px';
+    element: new SearchSectionBuilder()
+      .withDomElement(
+        $$(
+          'div',
+          {
+            className: 'CoveoPreferencesPanel'
+          },
+          $$('div', { className: 'CoveoResultsPreferences' })
+        )
+      )
+      .build(),
+    show: true,
+    toExecute: () => {
+      setMinHeightOnSearchInterface('300px');
     }
   },
   Settings: {
     show: true,
-    element: $$('div', undefined, `<div class="CoveoSettings"></div><div class="CoveoSearchbox"></div><div class="CoveoPreferencesPanel"></div><div class="CoveoShareQuery"></div><div class="CoveoExportToExcel"></div>`),
-    toExecute: ()=> {
-      $$(document.body).find('.CoveoSearchInterface').style.minHeight = '300px';
+    element: new SearchSectionBuilder()
+      .withComponent('CoveoShareQuery')
+      .withComponent('CoveoExportToExcel')
+      .withComponent('CoveoAdvancedSearch')
+      .build(),
+    toExecute: () => {
+      setMinHeightOnSearchInterface('300px');
     }
   },
   ShareQuery: {
     show: true,
-    element: $$('div', undefined, `<div class="CoveoSettings"></div><div class="CoveoSearchbox"></div><div class="CoveoShareQuery"></div>`),
-    toExecute: ()=> {
-      $$(document.body).find('.CoveoSearchInterface').style.minHeight = '300px';
+    element: new SearchSectionBuilder().withComponent('CoveoShareQuery').build(),
+    toExecute: () => {
+      setMinHeightOnSearchInterface('300px');
     }
   },
   SimpleFilter: {
     show: true,
-    element: $$('div', undefined, `<div class="CoveoSimpleFilter" data-field="@filetype" data-title="File Type"></div><div class="CoveoResultList"></div>`),
+    options: {
+      field: '@filetype',
+      title: 'File Type'
+    },
+    element: new SectionBuilder()
+      .withComponent('CoveoSimpleFilter')
+      .withComponent('CoveoResultList')
+      .build()
   },
   Sort: {
     show: true,
-    element: $$('div', undefined, `<div class="coveo-sort-section"><span class="CoveoSort" data-sort-criteria="relevancy" data-caption="Relevance"></span><span class="CoveoSort" data-sort-criteria="date descending,date ascending" data-caption="Date"></span></div><div class="CoveoResultList"></div>`),
-    toExecute: ()=> {
-      $$(document.body).find('.preview-container .CoveoSearchInterface').style.padding = '20px';
-      $$(document.body).on('buildingQuery', function(e, args) {
+    element: new SectionBuilder()
+      .withDomElement(
+        new SectionBuilder($$('div', { className: 'coveo-sort-section' }))
+          .withComponent(
+            'CoveoSort',
```

Comment:
General idea rather than a specific comment for this PR:
There are many strings for component names, attribute names, attribute values. Perhaps it would help to have them in a constants file to A) avoid potential spelling mistakes and have auto-autocomplete, B) help maintain when attribute names or their list of values change.